File: Week1/program3.c
/////////////////////////////////////////////////////////////////////////////////
//				Jay Ganesh
////////////////////////////////////////////////////////////////////////////////
////////////////////////////////////
// WAP to create dynmaic array and display only even elements from that Array
///////////////////////////////////
#include<stdio.h>
#include<stdlib.h>
////////////////////////////////////////////////////////
//Function Name:    Accept
//Description:      Accept dynamic array elements 
//Input:            int *,int 
//Output:           ---
//Date:             03/11/2022
//Author:           Manas Ohara
////////////////////////////////////////////////////////
void Accept(int *arr,int iSize)
{
	register int iCnt = 0;
	for(iCnt = 0 ; iCnt < iSize ; iCnt++ )
	{
		printf("Enter element %d : ",iCnt+1);
		scanf("%d",&arr[iCnt]);
	}
}
////////////////////////////////////////////////////////
//Function Name:    Display
//Description:      Display dynamic array elements 
//Input:            int *,int 
//Output:           ---
//Date:             03/11/2022
//Author:           Manas Ohara
////////////////////////////////////////////////////////
void Display(int *arr,int n)
{
	register int iCnt = 0;
	printf("Array elements are : \t");
	for(iCnt = 0 ; iCnt < n ; iCnt++)
	{
		printf("%d\t",arr[iCnt]);
	}
	printf("\n");
}
/////////////////////////////////////////////////////////
//Function Name:    DisplayEven
//Description:      Display dynamic array even elements 
//Input:            int *,int 
//Output:           ---
//Date:             03/11/2022
//Author:           Manas Ohara
////////////////////////////////////////////////////////
void DisplayEven(int *arr,int n)
{
	register int iCnt = 0;
	printf("Even Array elements are : \t");
	for(iCnt = 0 ; iCnt < n ; iCnt++)
	{
		if(arr[iCnt] % 2 == 0)
		{
			printf("%d\t",arr[iCnt]);
		}
	}
	printf("\n");
}
int main()
{
	int *arr = NULL;
	int iSize = 0;
	
	printf("Enter the size of an array : ");
	scanf("%d",&iSize);
	arr =(int *) malloc(sizeof(int)*iSize);
	if(arr == NULL )
	{
		printf("Unable to allocate the memory\n");
	}
	else
	{
		Accept(arr,iSize);
		Display(arr,iSize);
		DisplayEven(arr,iSize);
	}
	return 0;
}	

File: Week2/program3.java
/////////////////////////////////////////////////////////////////////////////////
//				Jay Ganesh
////////////////////////////////////////////////////////////////////////////////
import java.lang.*;
import java.util.*;
//////////////////////////////////////////////////////////////////////////////
// Selection Sort Algorithm
/////////////////////////////////////////////////////////////////////////////
class Array
{
	protected int arr[];	
	protected Scanner sobj = new Scanner(System.in);
	
	//Array Constructor
	Array(int iSize)
	{
		arr = new int[iSize];
	}
	
	/////////////////////////////////////////////////////////
	//Function Name:    Accept
	//Description:      accept array elements
	//Input:            ---
	//Output:           ---
	//Date:             10/11/2022
	//Author:           Manas Ohara
	////////////////////////////////////////////////////////
	protected void Accept()
	{
		System.out.println("Enter "+arr.length+" elements : ");
		for(int iCnt = 0 ; iCnt < arr.length ; iCnt++ )
		{
			arr[iCnt] = sobj.nextInt();
		}
	}
	
	/////////////////////////////////////////////////////////
	//Function Name:    Display
	//Description:      display array elements
	//Input:            ---
	//Output:           ---
	//Date:             10/11/2022
	//Author:           Manas Ohara
	////////////////////////////////////////////////////////
	protected void Display()
	{
		System.out.println("Arr : ");
		for(int iCnt = 0 ; iCnt < arr.length ; iCnt++ )
		{
			System.out.print(arr[iCnt]+" ");
		}
		System.out.println();
	}
}
class MyArray extends Array
{
	//constructor
	MyArray(int iSize)
	{
		//call to parent constructor
		super(iSize);
	}
	
	
	/////////////////////////////////////////////////////////
	//Function Name:    Sort
	//Description:      perform selection sort algorithm
	//Input:            ---
	//Output:           ---
	//Date:             10/11/2022
	//Author:           Manas Ohara
	////////////////////////////////////////////////////////
	protected void Sort()
	{
		for(int iCnt = 0 ; iCnt < arr.length ; iCnt++ )
		{
			int iMin_index = iCnt;
			for(int jCnt = iCnt+1 ; jCnt < arr.length ; jCnt++ )
			{
				if(arr[iMin_index] > arr[jCnt] )
				{
					iMin_index = jCnt;
				}
			}
			if(iMin_index != iCnt )
			{
				int temp = arr[iMin_index] ;
				arr[iMin_index] = arr[iCnt];
				arr[iCnt] = temp;
			}
		}
		System.out.println("After implementing selection sort arr : \n");
		super.Display();
	}

		
}
class program3
{
	public static void main(String argmnts[])
	{	
		System.out.println("Enter size of an array : ");
		Scanner sobj = new Scanner(System.in);
		int iSize = sobj.nextInt();
		MyArray obj = new MyArray(iSize);
		obj.Accept();
		obj.Display();
		obj.Sort();
	}
}
